dotCMS/src/test/java/com/dotmarketing/util/StringUtilsTest.java

```diff
@@ -315,4 +318,26 @@ public void testLowercaseStringExceptMatchingTokens(final String query,
 
     }
 
+    @DataProvider
+    public static Object[][] singleQuotedText() {
+        return new Object[][]{
+                {"'Tumblr' is an amazing app", 1},
+                {"Tumblr is an amazing 'app'", 1},
+                {"Tumblr is an 'amazing' app", 1 },
+                {"Tumblr is 'awesome' and 'amazing' ", 2},
+                {"Tumblr's users' are disappointed ", 0},
+                {"Tumblr's 'acquisition' complete but users' loyalty doubtful", 1},
+                {"Tumblr's la la la '123 456' text can be hard to parse. ", 0}, //<-- by design literals with blanks are excluded.
+                { "## Container: Blank Container\n"
+                + "## This is autogenerated code that cannot be changed\n"
+                + "#parseContainer('d71d56b4-0a8b-4bb2-be15-ffa5a23366ea','1539784124854')\n", 2}
+        };
+    }
+
+    @Test
+    @UseDataProvider("singleQuotedText")
+    public void testSingleQuotedText(String textBlock, int expectedResult){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'textBlock' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5002120)